service/history/outbound_queue_standby_task_executor.go

```diff
@@ -157,6 +157,15 @@ func (e *outboundQueueStandbyTaskExecutor) processTask(
 
 	historyResendInfo, err := actionFn(ctx)
 	if err != nil {
+		if errors.Is(err, consts.ErrTaskRetry) {
+			// If the error is retryable, it means the task hasn't been processed on the active side yet.
```

Comment:
nit: we don't know that for sure, could be just a delay in replication stack.

Reply:
Will  outbound task send a state sync on every retry from active?

Reply:
not today. We don't have a capability of doing state sync right now. and we will just wait for the replication task to be applied.
Once state-based replication is ready, all resend logic will be replace by state sync and sync will be done here as well.

Reply:
@yycptt Yes, the comment says that the "task hasn't been processed on the active side yet", and that "the most likely reason would be the destination is down".

I can add the reason "delay in replication stack" to the comment as well.